### ISSUE TITLE
[bug] BazelDeps fails to generate with libpng due to cpp_info.location

### ISSUE THREAD

**Issue by valgur** [conan-io/conan#17457]
### Describe the bug

Conan 2.10.2
Ubuntu 24.04

> ERROR: Error in generator 'BazelDeps': libpng/1.6.44: Cannot obtain 'location' for library 'png' in ['/home/martin/.conan2/p/b/libpn667f314ae558c/p/lib']. You can specify 'cpp_info.location' directly or report in github.com/conan-io/conan/issues if you think it should have been deduced correctly.

The only thing of note about the libpng recipe that I can see is that it's missing `package_type`, but adding it does not fix the issue.

### How to reproduce it

`conan install .` this conanfile.txt
```
[requires]
libpng/[>=1.6 <2]

[generators]
BazelDeps
```

**Comment by valgur** [comment]
The content of package_folder for libpng is:
```
├── conaninfo.txt
├── conanmanifest.txt
├── include
│   ├── libpng16
│   │   ├── pngconf.h
│   │   ├── png.h
│   │   └── pnglibconf.h
│   ├── pngconf.h
│   ├── png.h
│   └── pnglibconf.h
├── lib
│   ├── cmake
│   │   └── PNG
│   ├── libpng16.a
│   └── libpng.a -> libpng16.a
└── licenses
    └── LICENSE
```

My guess is that it's ignoring the `libpng.a`/`libpng.so` symlink for some reason.

**Comment by franramirez688** [comment]
Hi @valgur 

Thanks a lot for reporting this 😁 
Indeed, it was a bug, but this PR https://github.com/conan-io/conan/pull/17421 already solves it.
It'll be released in the next Conan 2.11.0 👏 

Closing the issue as solved, but do not hesitate to reopen if there is still any other problem.

**Comment by valgur** [comment]
Indeed, I can confirm that it's fixed in `develop2`. Thanks!
